### DISCUSSION TITLE
How To Fix NODE_ENV value in my environment.

### DISCUSSION THREAD

**Discussion by patilharsh03** [vercel/next.js#45666] [Help] [answered]
Whenever I run `npm run dev` i am getting this Warning message. How to fix it.

![image](https://user-images.githubusercontent.com/58598919/217294350-d2286bdb-aa3d-4f81-b002-1dc7f87c7cc4.png)

**Comment by icyJoseph** [comment, marked as answer]
That usually happens when you inject your own `process.env.NODE_ENV`.

Are you doing thing like `NODE_ENV=production next dev` in your `package.json`?

**Reply by patilharsh03** [reply]
This is my `package.json` I didn't even touch the file. I don't know what's the problem, every time I create the next app, these warnings came.

![image](https://user-images.githubusercontent.com/58598919/217512667-1007d24a-f47f-4899-9e31-61c50bbcaa44.png)

**Reply by ciruz** [reply]
Maybe you have a global NODE_ENV set, in your Windows System Control Panel.

**Reply by patilharsh03** [reply]
Ok. So how should I Fix it? any solutions

**Reply by icyJoseph** [reply]
Alright that's weird, but yeah perhaps you have a NODE_ENV setup globally in your environment. 

Not sure what OS, or shell you use, but try investigate how to print env vars on your machine/setup and see if there's a NODE_ENV configured.

Careful with posting back results, there might be sensitive data when logging env vars.

**Reply by icyJoseph** [reply]
@patilharsh03 if that's the case.

1. Think hard about how could it be that you ended up setting NODE_ENV? did you install something? follow a tutorial? etc, it is a bit concerning
2. Based on your shell or if its windows I guess the path (haven't used windows in a while), you need to unset the env var.

- https://stackoverflow.com/questions/13222724/command-line-to-remove-an-environment-variable-from-the-os-level-configuration
- https://www.tecmint.com/set-unset-environment-variables-in-linux/#2_Using_unset

You could also try to `echo` NODE_ENV from the terminal, see if it yields something.

**Reply by ciruz** [reply]
I guess you are on Windows? If so, open Power Shell (not cmd !!!) and run following command:

```powershell
get-childitem env:PATH
```

you should get a response, something like this:

![image](https://user-images.githubusercontent.com/987947/217532334-80fdf287-d0a3-4baf-bf2e-071a2fbeb6f0.png)
(Just to proof that the command works)

Now try the same command with ``NODE_ENV``, if there is no global ``NODE_ENV`` set, you should get an error, like so:

```powershell
get-childitem env:NODE_ENV
```
![image](https://user-images.githubusercontent.com/987947/217532600-e0633334-07fd-4a74-a8c8-0e4393f9f696.png)

If you get a response, you have a global ``NODE_ENV`` set and it's interfering with your next.js application.

**Reply by patilharsh03** [reply]
I got this response back when running `get-childitem env:NODE_ENV`

![image](https://user-images.githubusercontent.com/58598919/217533469-d022598d-0948-43ac-a526-3b990f9b07c4.png)

**Reply by icyJoseph** [reply]
Then that's it, you have `NODE_ENV` set as `dev`. That's very suspicious, do you remember setting that? I guess next up, figure out how to clear that, and hope that it doesn't break something else on your system 😅

**Reply by ciruz** [reply]
So you have a global ``NODE_ENV`` set then. Like @icyJoseph said I think you added one long time ago, following a tutorial or something.

If you think you don't need it, just remove it.

Press ``Windows-Key+R`` or open Power Shell again, now enter following command, to open the Environment Variables menu.

``rundll32 sysdm.cpl,EditEnvironmentVariables``

This opens the menu, it looks like this:

![image](https://user-images.githubusercontent.com/987947/217534558-d3bcd163-c99b-4b25-9f5b-489863a11f4c.png)

There should be a variable called ``NODE_ENV``, just delete it.

Close all dialogs, **also important, close all Power Shell / cmd windows**, even in vscode, and open them again (because they save the state of all variables temporary).

Now try the command once again in Power Shell (``get-childitem env:NODE_ENV``) to make sure the global variable is gone. If not, reboot your machine.

**Reply by patilharsh03** [reply]
Ok. I did as you told me and Restarted my device. Now the **warning has gone**. 

Thank You so much. It's very helpful of you both.
